account_payment_order/wizard/account_payment_line_create.py

```diff
@@ -0,0 +1,172 @@
+        comodel_name="account.journal", string="Journals Filter"
+    )
+    partner_ids = fields.Many2many(
+        comodel_name="res.partner",
+        string="Partners",
+        domain=[("parent_id", "=", False)],
+    )
+    target_move = fields.Selection(
+        selection=[("posted", "All Posted Entries"), ("all", "All Entries")],
+        string="Target Moves",
+    )
+    allow_blocked = fields.Boolean(string="Allow Litigation Move Lines")
+    invoice = fields.Boolean(string="Linked to an Invoice or Refund")
+    date_type = fields.Selection(
+        selection=[("due", "Due Date"), ("move", "Move Date")],
+        string="Type of Date Filter",
+        required=True,
+    )
+    due_date = fields.Date(string="Due Date")
+    move_date = fields.Date(string="Move Date", default=fields.Date.context_today)
+    payment_mode = fields.Selection(
+        selection=[("same", "Same"), ("same_or_null", "Same or Empty"), ("any", "Any")],
+        string="Payment Mode",
+    )
+    move_line_ids = fields.Many2many(
+        comodel_name="account.move.line", string="Move Lines"
+    )
+
+    @api.model
+    def default_get(self, field_list):
+        res = super(AccountPaymentLineCreate, self).default_get(field_list)
+        context = self.env.context
+        assert (
+            context.get("active_model") == "account.payment.order"
+        ), "active_model should be payment.order"
+        assert context.get("active_id"), "Missing active_id in context !"
+        order = self.env["account.payment.order"].browse(context["active_id"])
+        mode = order.payment_mode_id
+        res.update(
+            {
+                "journal_ids": mode.default_journal_ids.ids or False,
+                "target_move": mode.default_target_move,
+                "invoice": mode.default_invoice,
+                "date_type": mode.default_date_type,
+                "payment_mode": mode.default_payment_mode,
+                "order_id": order.id,
+            }
+        )
+        return res
+
+    def _prepare_move_line_domain(self):
+        self.ensure_one()
+        domain = [
+            ("reconciled", "=", False),
+            ("company_id", "=", self.order_id.company_id.id),
+        ]
+        if self.journal_ids:
+            domain += [("journal_id", "in", self.journal_ids.ids)]
+        if self.partner_ids:
+            domain += [("partner_id", "in", self.partner_ids.ids)]
+        if self.target_move == "posted":
+            domain += [("move_id.state", "=", "posted")]
+        if not self.allow_blocked:
+            domain += [("blocked", "!=", True)]
+        if self.date_type == "due":
+            domain += [
+                "|",
+                ("date_maturity", "<=", self.due_date),
+                ("date_maturity", "=", False),
+            ]
+        elif self.date_type == "move":
+            domain.append(("date", "<=", self.move_date))
+        if self.invoice:
+            domain.append(("invoice_id", "!=", False))
```

Comment:
invoice_id does not exist in account.move.line. The reference to the parent is now move_id